light_node/src/main.rs

```diff
@@ -310,14 +337,16 @@ fn block_on_actors(
     let chain_feeder_channel =
         ChainFeederChannel::actor(&actor_system).expect("Failed to create chain feeder channel");
 
+    let recorders = build_recorders(&env, &persistent_storage);
+
     // it's important to start ContextListener before ChainFeeder, because chain_feeder can trigger init_genesis which sends ContextActionMessage, and we need to process this action first
     let _ = ContextListener::actor(
         &actor_system,
         shell_channel.clone(),
         &persistent_storage,
+        recorders,
```

Comment:
@mat13mn 
if recorders are not needed elsewhere, I would directly call here: `build_recorders(&env, &persistent_storage)`,